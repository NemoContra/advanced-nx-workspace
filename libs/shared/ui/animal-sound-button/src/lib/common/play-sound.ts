import { Observable } from 'rxjs';

export const playSound = (src: string): Observable<void> => {
  return new Observable(() => {
    const audio = new Audio(src);
    audio.play();
    return () => audio.pause();
  });
};
