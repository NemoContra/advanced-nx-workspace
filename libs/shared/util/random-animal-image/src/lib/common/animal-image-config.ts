import { InjectionToken } from '@angular/core';

export interface AnimalImageConfig {
  cats: boolean;
  dogs: boolean;
}

export const defaultAnimalImageConfig: AnimalImageConfig = {
  cats: true,
  dogs: true,
};

export const ANIMAL_IMAGE_CONFIG = new InjectionToken<AnimalImageConfig>(
  'ANIMAL_IMAGE_CONFIG'
);
