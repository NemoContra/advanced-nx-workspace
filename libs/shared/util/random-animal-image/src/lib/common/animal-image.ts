export interface AnimalImage {
  src: string;
  alt: string;
}

export interface CatImage {
  file: string;
}

export interface DogImage {
  message: string;
  status: 'success' | 'failure';
}
