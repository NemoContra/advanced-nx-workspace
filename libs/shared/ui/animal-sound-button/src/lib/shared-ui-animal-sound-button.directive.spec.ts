import { createDirectiveFactory } from '@ngneat/spectator/jest';

import { SharedUiAnimalSoundButtonDirective } from './shared-ui-animal-sound-button.directive';
import { playSound } from './common/play-sound';

jest.mock('./common/play-sound', () => {
  const { of } = jest.requireActual('rxjs');
  return {
    playSound: jest.fn().mockReturnValue(of(undefined)),
  };
});

describe('SharedUiAnimalSoundButtonDirective ', () => {
  const createDirective = createDirectiveFactory(
    SharedUiAnimalSoundButtonDirective
  );

  describe.each([
    { animal: 'cat', sound: 'cat_purr_close' },
    { animal: 'dog', sound: 'dog_barking' },
  ])('$animal', ({ animal, sound }) => {
    afterEach(() => jest.resetAllMocks());

    it(`should play the audio ${sound}`, () => {
      const spectator = createDirective(
        `<button [flightWorkspaceSharedUiAnimalSoundButton]="animal">Click me!</button>`,
        {
          hostProps: { animal },
        }
      );

      expect(spectator.fixture).toMatchSnapshot();
      expect(playSound).not.toHaveBeenCalled();

      spectator.click('button');

      expect(playSound).toHaveBeenCalledTimes(1);
      expect(playSound).toHaveBeenCalledWith(
        `https://actions.google.com/sounds/v1/animals/${sound}.ogg`
      );
    });
  });
});
