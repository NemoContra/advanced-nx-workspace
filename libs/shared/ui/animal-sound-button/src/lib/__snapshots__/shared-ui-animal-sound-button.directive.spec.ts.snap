// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharedUiAnimalSoundButtonDirective  cat should play the audio cat_purr_close 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
  animal={[Function String]}
>
  <button
    title="I make animal sounds!"
  >
    Click me!
  </button>
</ng-component>
`;

exports[`SharedUiAnimalSoundButtonDirective  dog should play the audio dog_barking 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
  animal={[Function String]}
>
  <button
    title="I make animal sounds!"
  >
    Click me!
  </button>
</ng-component>
`;
